numpy/core/src/umath/ufunc_object.c

```diff
@@ -816,246 +917,164 @@ get_ufunc_arguments(PyUFuncObject *ufunc,
     if (kwds != NULL) {
         PyObject *key, *value;
         Py_ssize_t pos = 0;
-        while (PyDict_Next(kwds, &pos, &key, &value)) {
-            Py_ssize_t length = 0;
-            char *str = NULL;
-            int bad_arg = 1;
+        /*
+         * We store the outputs in order, so we can easily set them
+         * in our loop.  Note that if any of the out_* is NULL,
+         * the keyword will automatically be treated as unwanted.
+         */
+        kwarg = ufunc_kwargs;
+        /* Omit "out", since it is treated differently */
+        (++kwarg)->output = out_wheremask;
+        (++kwarg)->output = out_axes;
+        (++kwarg)->output = out_axis;
+        (++kwarg)->output = out_keepdims;
+        (++kwarg)->output = out_casting;
+        (++kwarg)->output = out_order;
+        (++kwarg)->output = &dtype;
+        (++kwarg)->output = out_subok;
+        (++kwarg)->output = out_typetup;
+        (++kwarg)->output = &sig;
+        (++kwarg)->output = out_extobj;
```

Comment:
This block isn't threadsafe

Reply:
Oops, I had not thought of that... I actually had a parallel array before, and at the time it seemed less nice. But reverting, I actually think it is much nicer, so I adopted it.